### DISCUSSION TITLE
Cannot get turbo_frame_tag to print

### DISCUSSION THREAD

**Discussion by tillcarlos** [yippee-fun/phlex#217] [Help] [answered]
Hey there - great gem. 

Plex gave us a speed-up of at least 60% compared to calling hundreds of partial renders (before you jump in: I know, it shouldn't be done that way in the first place).

A problem I'm facing now: print a `turbo-frame`, but couldn't figure it out yet:

```ruby
module Views
  class TableCell < ApplicationView
    delegate :turbo_frame_tag, to: :helpers


    def template
        # super weird. it should work: https://github.com/hotwired/turbo-rails/blob/main/app/helpers/turbo/frames_helper.rb 
        turbo_frame_tag id: "foo" do 
          p "content"
        end
    end
end
```

it prints the content, but not the tag.

I guess that's due to the turbo_frame_tag itself - but maybe there's a workaround I'm not seeing yet?

**Comment by joeldrapper** [comment, marked as answer]
Hey @tillcarlos, in your `ApplicationView`, you can register a custom element with `register_element :turbo_frame`. Then just use the `turbo_frame` element method rather than the `turbo_frame_tag` helper.

We should address this more directly in the framework and documentation, as it's really confusing as it is. The problem with the Rails tag helpers is they return a String of the HTML, they don't append directly to the output buffer. That makes it quite tricky to integrate with them, since we have no way to capture the return value of multiple statements unless they're passed to a method. By way of example, you could probably get `raw turbo_frame_tag` to work.

**Reply by tillcarlos** [reply]
This is amazing and worked right away. Thanks!!

Yeah, I guessed that was due to some buffer issue - it's just hard to see exactly how this works just guessing from the documentation.

All good though. Thanks for the great work!

**Comment by joeldrapper** [comment]
> Plex gave us a speed-up of at least 60% compared to calling hundreds of partial renders (before you jump in: I know, it shouldn't be done that way in the first place).

Why not? I use Tailwind and make a new Phlex::View practically every time I would have otherwise made a semantic CSS class.

**Reply by tillcarlos** [reply]
I meant: when you use a `render 'partial'` method you should not call this in a loop too often. Best practice here is to send a collection into the render method. For some reason the render method is way slower than using phlex. so that's good for us now.

ref: https://thoughtbot.com/blog/rendering-collections-in-rails

**Reply by joeldrapper** [reply]
Oh I see. 👍

Can't wait to show you what I’m working on for rendering collections in Phlex!
